src/xcode/ENA/ENA/Source/Scenes/ContactDiary/StoreV2/ContactDiaryStoreV2.swift

```diff
@@ -0,0 +1,995 @@
+				return
+			}
+
+			let updateDiaryDaysResult = updateDiaryDays(with: database)
+			guard case .success = updateDiaryDaysResult else {
+				logLastErrorCode(from: database)
+				result = .failure(dbError(from: database))
+				return
+			}
+
+			result = .success(())
+		}
+
+		guard let _result = result else {
+			fatalError("[ContactDiaryStore] Result should not be nil.")
+		}
+
+		return _result
+	}
+
+	private func dropTables() -> DiaryStoringVoidResult {
+		var result: DiaryStoringVoidResult?
+
+		databaseQueue.inDatabase { database in
+			let sql = """
+					PRAGMA journal_mode=OFF;
+					DROP TABLE Location;
+					DROP TABLE LocationVisit;
+					DROP TABLE ContactPerson;
+					DROP TABLE ContactPersonEncounter;
+					VACUUM;
+				"""
+
+			guard database.executeStatements(sql) else {
+				logLastErrorCode(from: database)
+				result = .failure(dbError(from: database))
+				return
+			}
+
+			result = .success(())
+		}
+
+		guard let _result = result else {
+			fatalError("[ContactDiaryStore] Result should not be nil.")
+		}
+
+		return _result
+	}
+
+
+	private func logLastErrorCode(from database: FMDatabase) {
+		Log.error("[ContactDiaryStore] (\(database.lastErrorCode())) \(database.lastErrorMessage())", log: .localData)
+	}
+
+	private func dbError(from database: FMDatabase) -> DiaryStoringError {
+		let dbError = SQLiteErrorCode(rawValue: database.lastErrorCode()) ?? SQLiteErrorCode.unknown
+		return .database(dbError)
+	}
+}
+
+// MARK: Creation
+
+extension ContactDiaryStoreV2 {
+
+	static func make() -> ContactDiaryStoreV2 {
+		Log.info("[ContactDiaryStore] Trying to create contact diary store...", log: .localData)
+
+		if let store = ContactDiaryStoreV2() {
+			Log.info("[ContactDiaryStore] Successfully created contact diary store", log: .localData)
+			return store
+		}
+
+		Log.info("[ContactDiaryStore] Failed to create contact diary store. Try to rescue it...", log: .localData)
+
+		// The database could not be created – To the rescue!
+		// Remove the database file and try to init the store a second time.
+		try? FileManager.default.removeItem(at: ContactDiaryStoreV2.storeDirectoryURL)
+
+		if let secondTryStore = ContactDiaryStoreV2() {
+			Log.info("[ContactDiaryStore] Successfully rescued contact diary store", log: .localData)
+			return secondTryStore
+		} else {
+			Log.info("[ContactDiaryStore] Failed to rescue contact diary store.", log: .localData)
+			fatalError("[ContactDiaryStore] Could not create contact diary store after second try.")
+		}
+	}
+
+//	convenience init?() {
+//		guard let databaseQueue = FMDatabaseQueue(path: ContactDiaryStoreV2.storeURL.path) else {
+//			Log.error("[ContactDiaryStore] Failed to create FMDatabaseQueue.", log: .localData)
+//			return nil
+//		}
+//
+//		let schema = ContactDiaryStoreSchemaV1(
+//			databaseQueue: databaseQueue
+//		)
+//
+//		self.init(
+//			databaseQueue: databaseQueue,
+//			schema: schema,
+//			key: ContactDiaryStoreV2.encryptionKey
+//		)
+//	}
```

Comment:
Needed?

```suggestion
```

Reply:
nope, forgot to delete it before committing, will push a new commit soon